File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        front = []
        rear = []
        front.append(nums[0])
        rear.append(nums[len(nums) - 1])
        for i in range(1, len(nums)):
            front.append(front[-1] + nums[i])
        print(front)
        for i in range(len(nums) - 2, -1, -1):
            rear = [rear[0] + nums[i]] + rear
        print(rear)
        for i in range(len(nums)):
            if rear[i] == front[i]:
                return i
        return -1

        

File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        ArrayList<Integer> front = new ArrayList<>();
        front.add(nums[0]);
        ArrayList<Integer> rear = new ArrayList<>();
        rear.add(nums[nums.length - 1]);
        for(int i = 1; i < nums.length; i++){
            front.add(front.get(i - 1) + nums[i]);
        }
        for(int i = nums.length - 2; i >= 0; i--){
            rear.add(0, rear.get(0) + nums[i]);
        }
        for(int i =0; i<nums.length; i++){
            if (front.get(i).equals(rear.get(i))) {
                return i;
            }
        }
        return -1;
    }
}